tools/network/telemetryURIUpdateService.go

```diff
@@ -44,27 +71,62 @@ func StartTelemetryURIUpdateService(interval time.Duration, cfg config.Local, ge
 			select {
 			case <-ticker.C:
 				updateTelemetryURI()
-			case <-abort:
+			case <-t.abort:
 				return
 			}
 		}
 	}()
 }
 
-func lookupTelemetryEndpoint(cfg config.Local, genesisNetwork protocol.NetworkID, log logging.Logger) string {
-	bootstrapArray := cfg.DNSBootstrapArray(genesisNetwork)
+func (t *telemetryURIUpdater) lookupTelemetryURL() (url *url.URL) {
+	bootstrapArray := t.cfg.DNSBootstrapArray(t.genesisNetwork)
 	bootstrapArray = append(bootstrapArray, "default.algodev.network")
 	for _, bootstrapID := range bootstrapArray {
-		addrs, err := ReadFromSRV("telemetry", bootstrapID, cfg.FallbackDNSResolverAddress)
+		addrs, err := t.srvReader.readFromSRV("tls", bootstrapID)
 		if err != nil {
-			log.Infof("An issue occurred reading telemetry entry for '%s': %v", bootstrapID, err)
+			t.log.Infof("An issue occurred reading telemetry entry for '_telemetry._tls.%s': %v", bootstrapID, err)
 		} else if len(addrs) == 0 {
-			log.Infof("No telemetry entry for: '%s'", bootstrapID)
+			t.log.Infof("No telemetry entry for: '_telemetry._tls.%s'", bootstrapID)
 		} else {
-			return addrs[0]
+			for _, addr := range addrs {
+				// the add that we received from ReadFromSRV contains host:port, we need to prefix that with the schema. since it's the tls, we want to use https.
+				url, err = url.Parse("https://" + addr)
+				if err != nil {
+					t.log.Infof("a telemetry endpoint '%s' was retrieved for '_telemerty._tls.%s'. This does not seems to be a valid endpoint and will be ignored(%v).", addr, bootstrapID, err)
+					continue
+				}
+				return url
+			}
+		}
+
+		addrs, err = t.srvReader.readFromSRV("tcp", bootstrapID)
+		if err != nil {
+			t.log.Infof("An issue occurred reading telemetry entry for '_telemetry._tcp.%s': %v", bootstrapID, err)
+		} else if len(addrs) == 0 {
+			t.log.Infof("No telemetry entry for: '_telemetry._tcp.%s'", bootstrapID)
+		} else {
+			for _, addr := range addrs {
+				if strings.HasPrefix(addr, "https://") {
+					// the add that we received from ReadFromSRV should contains host:port. however, in some cases, it migth contains a https prefix, where we want to take it as is.
```

Comment:
Some typos in comments. add -> url in most of them, in this comment I think you mean "should contains" -> "should not contain"